pkg/cloudprovider/providers/openstack/openstack_loadbalancer.go

```diff
@@ -863,7 +868,29 @@ func (lbaas *LbaasV2) EnsureLoadBalancer(ctx context.Context, clusterName string
 		return nil, fmt.Errorf("no ports provided to openstack load balancer")
 	}
 
+	internalAnnotation := false
+	var lbClass *LBClass
+	var err error
+	class := getStringFromServiceAnnotation(apiService, ServiceAnnotationLoadBalancerClass, "")
+	if class != "" {
+		lbClass = lbaas.opts.LBClasses[class]
+		if lbClass == nil {
+			return nil, fmt.Errorf("invalid loadbalancer class %q", class)
+		}
+		internalAnnotation = lbClass.SubnetID != ""
```

Comment:
If I understand it correctly, the global option `lbaas.opts.InternalLB` will be ignored when a lb class is set.
Instead the controller will create a internal lb when a class with a defined subnet ID is set.
Not sure if this is the behaviour that you desire here, but it sounds a bit unintuitive to me...

Also, when a class sets a subnet ID and a floating network ID, wouldn't that prevent the creation of the floating ip with your code?

Reply:
When an LBclass is requested via service annotation but can't be found in the controller configuration, an error will be thrown.

Reply:
Ok, we addressed this issue. Throwing an error now if the LBclass is misconfigured.